Jenkinsfile

```diff
@@ -105,5 +115,36 @@ pipeline {
                 }
             }
         }
+
+        stage("Build libcpptraj docker container") {
+            agent { label "linux && docker" }
+
+            environment {
+                DOCKER_IMAGE_TAG = "${env.BRANCH_NAME == "master" ? "master" : env.ghprbActualCommit}"
+            }
+
+            steps {
+                echo "Building and pushing ambermd/libcpptraj:${env.DOCKER_IMAGE_TAG}"
```

Comment:
How many images we will keep? (each image costs ~700M).

Reply:
Yea, I don't have a great answer for this.  Ideally the only one I want to keep long-term is the `master` branch.  The PR-specific tags I'd like to get rid of, but I don't have a good way other than manually purging them periodically.  It's also unclear to me how long some of those PR-specific images will be useful, particularly if it breaks pytraj.

It isn't all that bad, though.  Each image involves ~700M of download, but the images themselves should only require a small amount of extra storage.

Docker images are composed of multiple layers that are put together to construct a full container.  Each line of the Dockerfile will generate a new layer (which is a function both of the layers it is built on top of as well as the command that creates that new layer).

Since the bulk of the image is the same for *every* `ambermd/libcpptraj` image -- specifically the part that comes at the top -- those layers will be the same for every `libcpptraj` image and thus stored only once.

The only layers that change are the ones that contain the compiled cpptraj source code.  So all these images won't actually take up that much space on Dockerhub.